app/controllers/profile.js

```diff
@@ -14,34 +15,176 @@ export default class ProfileController extends Controller {
+      icon_url: 'icons/user.svg',
+      showError: false,
+      errorMessage: 'Designation is required',
     },
     {
-      id: 'yoe',
-      label: 'Years of Experience',
+      id: 'linkedin_id',
+      label: 'LinkedIn ID*',
+      type: 'text',
+      required: true,
+      placeholder: 'e.g johndoe',
+      icon_url: 'icons/linkedin.svg',
+      showError: false,
+      errorMessage: 'Linkedin handle is required',
     },
     {
-      id: 'github_id',
-      label: 'Github ID',
+      id: 'twitter_id',
+      label: 'Twitter Username*',
+      type: 'text',
+      required: true,
+      placeholder: 'e.g johndoe',
+      icon_url: 'icons/twitter.svg',
+      showError: false,
+      errorMessage: 'Twitter handle is required',
     },
     {
-      id: 'github_display_name',
-      label: 'Github Display Name',
+      id: 'website',
+      label: 'Personal Website',
+      type: 'text',
+      required: false,
+      placeholder: 'e.g mysite.com',
+      icon_url: 'icons/website.svg',
+      showError: false,
+      errorMessage: '',
     },
   ];
 
+  @action handleFieldChange(name, value) {
+    set(this.formData, name, value);
+
+    const anyErrors = this.fields.map((field) => {
+      let hasError = false;
+
+      if (field.required && this.formData[field.id] === '') {
+        hasError = true;
+      } else if (field.validator) {
+        if (!field.validator(this.formData[field.id])) {
+          hasError = true;
+        }
+      }
+      return hasError;
+    });
+    this.isSubmitDisabled = anyErrors.filter(Boolean).length;
+  }
+
+  @action handleFieldValidation(id, isValid) {
+    const index = this.fields.findIndex((field) => field.id === id);
+    if (isValid) {
+      set(this.fields[index], 'showError', false);
+    } else {
+      set(this.fields[index], 'showError', true);
+    }
+  }
+
+  @action emailValidator(email) {
+    if (typeof email !== 'string') return false;
+    const pattern =
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+    return pattern.test(email) ? true : false;
+  }
+
+  removeEmptyFields(reqObject) {
+    const objectRequested = reqObject;
+    for (const field in objectRequested) {
+      if (!objectRequested[field]) {
+        delete objectRequested[field];
+      } else if (field === 'yoe') {
+        objectRequested[field] = parseInt(objectRequested[field]);
+      }
+    }
+    return objectRequested;
+  }
+
+  @action async handleSubmit(e) {
+    e.preventDefault();
+    const cleanReqObject = this.removeEmptyFields(this.formData);
+    try {
+      const response = await fetch(`${BASE_URL}/users/self`, {
+        method: 'PATCH',
+        body: JSON.stringify(cleanReqObject),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
+
+      const { status } = response;
+      if (status !== 204) {
+        this.toast.error(
+          'Something went wrong. Please check console errors.',
```

Comment:
Can we please display the toast on success with timeout of 1 second and whatever the default is present in codebase as well ??
**Content :** Updated details successfully

Reply:
This leads to one more problem that whenever I want to update any details, due to validations being put on email `Submit` button doesn't get enabled until the email is correctly filled

[Screencast from 2023-05-11 00-37-54.webm](https://github.com/Real-Dev-Squad/website-my/assets/78433013/45510bdd-5729-4041-93c0-02231d5b55ab)